<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>ATACseqQC Guide • ATACseqQC</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="ATACseqQC Guide">
<meta property="og:description" content="ATACseqQC">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ATACseqQC</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.15.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/ATACseqQC.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ATACseqQC_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>ATACseqQC Guide</h1>
                        <h4 class="author">Jianhong Ou, Haibo Liu, Jun Yu, Michelle Kelliher, Lucio Castilla, Nathan Lawson, Lihua Julie Zhu</h4>
            
            <h4 class="date">2 April 2021</h4>
      
      
      <div class="hidden name"><code>ATACseqQC.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Assay for Transposase-Accessible Chromatin using sequencing (ATAC-seq) is an alternative or complementary technique to MNase-seq, DNase-seq, and FAIRE-seq for chromatin accessibility analysis. The results obtained from ATAC-seq are similar to those from DNase-seq and FAIRE-seq. ATAC-seq is gaining popularity because it does not require cross-linking, has higher signal to noise ratio, requires a much smaller amount of biological material and is faster and easier to perform, compared to other techniques.<span class="citation"><sup>1</sup></span></p>
<p>To help researchers quickly assess the quality of ATAC-seq data, we have developed the ATACseqQC package for easily making diagnostic plots following the published guidelines.<span class="citation"><sup>1</sup></span> In addition, it has functions to preprocess ATACseq data for subsequent peak calling.</p>
</div>
<div id="quick-start" class="section level1">
<h1 class="hasAnchor">
<a href="#quick-start" class="anchor"></a>Quick start</h1>
<p>Here is an example using ATACseqQC with a subset of published ATAC-seq data<span class="citation"><sup>1</sup></span>. Currently, only bam input file format is supported.</p>
<p>First install ATACseqQC and other packages required to run the examples. Please note that the example dataset used here is from human. To run analysis with dataset from a different species or differnt assembly, please install the corresponding BSgenome, TxDb and phastCons. For example, to analyze mouse data aligned to mm10, please install BSgenome.Mmusculus.UCSC.mm10, TxDb.Mmusculus.UCSC.mm10.knownGene and phastCons60way.UCSC.mm10. Please note that phstCons60way.UCSC.mm10 is optional, which can be obtained according to the vignettes of GenomicScores.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BiocManager</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ATACseqQC"</span>, <span class="st">"ChIPpeakAnno"</span>, <span class="st">"MotifDb"</span>, <span class="st">"GenomicAlignments"</span>,
           <span class="st">"BSgenome.Hsapiens.UCSC.hg19"</span>, <span class="st">"TxDb.Hsapiens.UCSC.hg19.knownGene"</span>,
           <span class="st">"phastCons100way.UCSC.hg19"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## load the library</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ATACseqQC</span><span class="op">)</span>
<span class="co">## input the bamFile from the ATACseqQC package </span>
<span class="va">bamfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GL1.bam"</span>, 
                        package<span class="op">=</span><span class="st">"ATACseqQC"</span>, mustWork<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">bamfile.labels</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">".bam"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span><span class="op">(</span><span class="va">bamfile</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div id="igv-snapshot" class="section level2">
<h2 class="hasAnchor">
<a href="#igv-snapshot" class="anchor"></a>IGV snapshot</h2>
<p>Source code of IGVSnapshot function is available in extdata folder. To call the function, please try</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"IGVSnapshot.R"</span>, package <span class="op">=</span> <span class="st">"ATACseqQC"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="estimate-the-library-complexity" class="section level2">
<h2 class="hasAnchor">
<a href="#estimate-the-library-complexity" class="anchor"></a>Estimate the library complexity</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#bamQC(bamfile, outPath=NULL)</span>
<span class="fu"><a href="../reference/estimateLibComplexity.html">estimateLibComplexity</a></span><span class="op">(</span><span class="fu"><a href="../reference/readsDupFreq.html">readsDupFreq</a></span><span class="op">(</span><span class="va">bamfile</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="ATACseqQC_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
<div id="fragment-size-distribution" class="section level2">
<h2 class="hasAnchor">
<a href="#fragment-size-distribution" class="anchor"></a>Fragment size distribution</h2>
<p>First, there should be a large proportion of reads with less than 100 bp, which represents the nucleosome-free region. Second, the fragment size distribution should have a clear periodicity, which is evident in the inset figure, indicative of nucleosome occupacy (present in integer multiples).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## generate fragement size distribution</span>
<span class="va">fragSize</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fragSizeDist.html">fragSizeDist</a></span><span class="op">(</span><span class="va">bamfile</span>, <span class="va">bamfile.labels</span><span class="op">)</span></code></pre></div>
<p><img src="ATACseqQC_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
<div id="nucleosome-positioning" class="section level2">
<h2 class="hasAnchor">
<a href="#nucleosome-positioning" class="anchor"></a>Nucleosome positioning</h2>
<div id="adjust-the-read-start-sites" class="section level3">
<h3 class="hasAnchor">
<a href="#adjust-the-read-start-sites" class="anchor"></a>Adjust the read start sites</h3>
<p>Tn5 transposase has been shown to bind as a dimer and inserts two adaptors into accessible DNA locations separated by 9 bp.<span class="citation"><sup>2</sup></span></p>
<p>Therefore, for downstream analysis, such as peak-calling and footprinting, all reads in input bamfile need to be shifted. The function <code>shiftGAlignmentsList</code> can be used to shift the reads. By default, all reads aligning to the positive strand are offset by +4bp, and all reads aligning to the negative strand are offset by -5bp.<span class="citation"><sup>1</sup></span></p>
<p>The adjusted reads will be written into a new bamfile for peak calling or footprinting.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## bamfile tags to be read in</span>
<span class="va">possibleTag</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"integer"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"AM"</span>, <span class="st">"AS"</span>, <span class="st">"CM"</span>, <span class="st">"CP"</span>, <span class="st">"FI"</span>, <span class="st">"H0"</span>, <span class="st">"H1"</span>, <span class="st">"H2"</span>, 
                                <span class="st">"HI"</span>, <span class="st">"IH"</span>, <span class="st">"MQ"</span>, <span class="st">"NH"</span>, <span class="st">"NM"</span>, <span class="st">"OP"</span>, <span class="st">"PQ"</span>, <span class="st">"SM"</span>,
                                <span class="st">"TC"</span>, <span class="st">"UQ"</span><span class="op">)</span>, 
                 <span class="st">"character"</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"BC"</span>, <span class="st">"BQ"</span>, <span class="st">"BZ"</span>, <span class="st">"CB"</span>, <span class="st">"CC"</span>, <span class="st">"CO"</span>, <span class="st">"CQ"</span>, <span class="st">"CR"</span>,
                               <span class="st">"CS"</span>, <span class="st">"CT"</span>, <span class="st">"CY"</span>, <span class="st">"E2"</span>, <span class="st">"FS"</span>, <span class="st">"LB"</span>, <span class="st">"MC"</span>, <span class="st">"MD"</span>,
                               <span class="st">"MI"</span>, <span class="st">"OA"</span>, <span class="st">"OC"</span>, <span class="st">"OQ"</span>, <span class="st">"OX"</span>, <span class="st">"PG"</span>, <span class="st">"PT"</span>, <span class="st">"PU"</span>,
                               <span class="st">"Q2"</span>, <span class="st">"QT"</span>, <span class="st">"QX"</span>, <span class="st">"R2"</span>, <span class="st">"RG"</span>, <span class="st">"RX"</span>, <span class="st">"SA"</span>, <span class="st">"TS"</span>,
                               <span class="st">"U2"</span><span class="op">)</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Rsamtools">Rsamtools</a></span><span class="op">)</span>
<span class="va">bamTop100</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/scanBam.html">scanBam</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/BamFile-class.html">BamFile</a></span><span class="op">(</span><span class="va">bamfile</span>, yieldSize <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>,
                     param <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/ScanBamParam-class.html">ScanBamParam</a></span><span class="op">(</span>tag<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">possibleTag</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">tag</span>
<span class="va">tags</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">bamTop100</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lengths.html">lengths</a></span><span class="op">(</span><span class="va">bamTop100</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span><span class="op">]</span>
<span class="va">tags</span></code></pre></div>
<pre><code>##    integer2   integer13 character16 
##        "AS"        "NM"        "MD"</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## files will be output into outPath</span>
<span class="va">outPath</span> <span class="op">&lt;-</span> <span class="st">"splited"</span>
<span class="fu"><a href="https://rdrr.io/r/base/files2.html">dir.create</a></span><span class="op">(</span><span class="va">outPath</span><span class="op">)</span>
<span class="co">## shift the coordinates of 5'ends of alignments in the bam file</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Hsapiens.UCSC.hg19</span><span class="op">)</span>
<span class="va">seqlev</span> <span class="op">&lt;-</span> <span class="st">"chr1"</span> <span class="co">## subsample data for quick run</span>
<span class="va">which</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">Hsapiens</span><span class="op">)</span><span class="op">[</span><span class="va">seqlev</span><span class="op">]</span>, <span class="st">"GRanges"</span><span class="op">)</span>
<span class="va">gal</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readBamFile.html">readBamFile</a></span><span class="op">(</span><span class="va">bamfile</span>, tag<span class="op">=</span><span class="va">tags</span>, which<span class="op">=</span><span class="va">which</span>, asMates<span class="op">=</span><span class="cn">TRUE</span>, bigFile<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">shiftedBamfile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">outPath</span>, <span class="st">"shifted.bam"</span><span class="op">)</span>
<span class="va">gal1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/shiftGAlignmentsList.html">shiftGAlignmentsList</a></span><span class="op">(</span><span class="va">gal</span>, outbam<span class="op">=</span><span class="va">shiftedBamfile</span><span class="op">)</span></code></pre></div>
</div>
<div id="promotertranscript-body-pt-score" class="section level3">
<h3 class="hasAnchor">
<a href="#promotertranscript-body-pt-score" class="anchor"></a>Promoter/Transcript body (PT) score</h3>
<p>PT score is calculated as the coverage of promoter divided by the coverage of its transcript body. PT score will show if the signal is enriched in promoters.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span>
<span class="va">txs</span> <span class="op">&lt;-</span> <span class="fu">transcripts</span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span>
<span class="va">pt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/PTscore.html">PTscore</a></span><span class="op">(</span><span class="va">gal1</span>, <span class="va">txs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">pt</span><span class="op">$</span><span class="va">log2meanCoverage</span>, <span class="va">pt</span><span class="op">$</span><span class="va">PT_score</span>, 
     xlab<span class="op">=</span><span class="st">"log2 mean coverage"</span>,
     ylab<span class="op">=</span><span class="st">"Promoter vs Transcript"</span><span class="op">)</span></code></pre></div>
<p><img src="ATACseqQC_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div id="nucleosome-free-regions-nfr-score" class="section level3">
<h3 class="hasAnchor">
<a href="#nucleosome-free-regions-nfr-score" class="anchor"></a>Nucleosome Free Regions (NFR) score</h3>
<p>NFR score is a ratio between cut signal adjacent to TSS and that flanking the corresponding TSS. Each TSS window of 400 bp is first divided into 3 sub-regions: the most upstream 150 bp (n1), the most downstream of 150 bp (n2), and the middle 100 bp (nf). Then the number of fragments with 5’ ends overlapping each region are calculated for each TSS. The NFR score for each TSS is calculated as NFR-score = log2(nf) - log2((n1+n2)/2). A plot can be generated with the NFR scores as Y-axis and the average signals of 400 bp window as X-axis, very like a MA plot for gene expression data.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">nfr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/NFRscore.html">NFRscore</a></span><span class="op">(</span><span class="va">gal1</span>, <span class="va">txs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">nfr</span><span class="op">$</span><span class="va">log2meanCoverage</span>, <span class="va">nfr</span><span class="op">$</span><span class="va">NFR_score</span>, 
     xlab<span class="op">=</span><span class="st">"log2 mean coverage"</span>,
     ylab<span class="op">=</span><span class="st">"Nucleosome Free Regions score"</span>,
     main<span class="op">=</span><span class="st">"NFRscore for 200bp flanking TSSs"</span>,
     xlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">10</span>, <span class="fl">0</span><span class="op">)</span>, ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">5</span>, <span class="fl">5</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="ATACseqQC_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div id="transcription-start-site-tss-enrichment-score" class="section level3">
<h3 class="hasAnchor">
<a href="#transcription-start-site-tss-enrichment-score" class="anchor"></a>Transcription Start Site (TSS) Enrichment Score</h3>
<p>TSS enrichment score is a raio between aggregate distribution of reads centered on TSSs and that flanking the corresponding TSSs. TSS score = the depth of TSS (each 100bp window within 1000 bp each side) / the depth of end flanks (100bp each end). TSSE score = max(mean(TSS score in each window)). TSS enrichment score is calculated according to the definition at <a href="https://www.encodeproject.org/data-standards/terms/#enrichment">https://www.encodeproject.org/data-standards/terms/#enrichment</a>. Transcription start site (TSS) enrichment values are dependent on the reference files used; cutoff values for high quality data are listed in the following table from <a href="https://www.encodeproject.org/atac-seq/">https://www.encodeproject.org/atac-seq/</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tsse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TSSEscore.html">TSSEscore</a></span><span class="op">(</span><span class="va">gal1</span>, <span class="va">txs</span><span class="op">)</span>
<span class="va">tsse</span><span class="op">$</span><span class="va">TSSEscore</span></code></pre></div>
<pre><code>## [1] 26.87021</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="fl">100</span><span class="op">*</span><span class="op">(</span><span class="op">-</span><span class="fl">9</span><span class="op">:</span><span class="fl">10</span><span class="op">-</span><span class="fl">.5</span><span class="op">)</span>, <span class="va">tsse</span><span class="op">$</span><span class="va">values</span>, type<span class="op">=</span><span class="st">"b"</span>, 
     xlab<span class="op">=</span><span class="st">"distance to TSS"</span>,
     ylab<span class="op">=</span><span class="st">"aggregate TSS score"</span><span class="op">)</span></code></pre></div>
<p><img src="ATACseqQC_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div id="split-reads" class="section level3">
<h3 class="hasAnchor">
<a href="#split-reads" class="anchor"></a>Split reads</h3>
<p>The shifted reads will be split into different bins, namely nucleosome free, mononucleosome, dinucleosome, and trinucleosome. Shifted reads that do not fit into any of the above bins will be discarded. Splitting reads is a time-consuming step because we are using random forest to classify the fragments based on fragment length, GC content and conservation scores.<span class="citation"><sup>3</sup></span></p>
<p>By default, we assign the top 10% of short reads (reads below 100_bp) as nucleosome-free regions and the top 10% of intermediate length reads as (reads between 180 and 247 bp) mononucleosome. This serves as the training set to classify the rest of the fragments using random forest. The number of the tree will be set to 2 times of square root of the length of the training set.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">phastCons100way.UCSC.hg19</span><span class="op">)</span>
<span class="co">## run program for chromosome 1 only</span>
<span class="va">txs</span> <span class="op">&lt;-</span> <span class="va">txs</span><span class="op">[</span><span class="fu">seqnames</span><span class="op">(</span><span class="va">txs</span><span class="op">)</span> <span class="op">%in%</span> <span class="st">"chr1"</span><span class="op">]</span>
<span class="va">genome</span> <span class="op">&lt;-</span> <span class="va">Hsapiens</span>
<span class="co">## split the reads into NucleosomeFree, mononucleosome, </span>
<span class="co">## dinucleosome and trinucleosome.</span>
<span class="co">## and save the binned alignments into bam files.</span>
<span class="va">objs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/splitGAlignmentsByCut.html">splitGAlignmentsByCut</a></span><span class="op">(</span><span class="va">gal1</span>, txs<span class="op">=</span><span class="va">txs</span>, genome<span class="op">=</span><span class="va">genome</span>, outPath <span class="op">=</span> <span class="va">outPath</span>,
                              conservation<span class="op">=</span><span class="va">phastCons100way.UCSC.hg19</span><span class="op">)</span>
<span class="co">## list the files generated by splitGAlignmentsByCut.</span>
<span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span><span class="op">(</span><span class="va">outPath</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "dinucleosome.bam"       "dinucleosome.bam.bai"   "inter1.bam"            
##  [4] "inter1.bam.bai"         "inter2.bam"             "inter2.bam.bai"        
##  [7] "inter3.bam"             "inter3.bam.bai"         "mononucleosome.bam"    
## [10] "mononucleosome.bam.bai" "NucleosomeFree.bam"     "NucleosomeFree.bam.bai"
## [13] "others.bam"             "others.bam.bai"         "shifted.bam"           
## [16] "shifted.bam.bai"        "trinucleosome.bam"      "trinucleosome.bam.bai"</code></pre>
<p>You can also perform shifting, splitting and saving in one step by calling <code>splitBam</code>.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">objs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/splitBam.html">splitBam</a></span><span class="op">(</span><span class="va">bamfile</span>, tags<span class="op">=</span><span class="va">tags</span>, outPath<span class="op">=</span><span class="va">outPath</span>,
                 txs<span class="op">=</span><span class="va">txs</span>, genome<span class="op">=</span><span class="va">genome</span>,
                 conservation<span class="op">=</span><span class="va">phastCons100way.UCSC.hg19</span><span class="op">)</span></code></pre></div>
<p>Conservation is an optional parameter. If you do not have the conservation score or you would like to simply split the bam files using the fragment length, then you will just need to run the command without providing the conservation argument. Without setting the conservation parameter, it will run much faster.</p>
</div>
<div id="heatmap-and-coverage-curve-for-nucleosome-positions" class="section level3">
<h3 class="hasAnchor">
<a href="#heatmap-and-coverage-curve-for-nucleosome-positions" class="anchor"></a>Heatmap and coverage curve for nucleosome positions</h3>
<p>By averaging the signal across all active TSSs, we should observe that nucleosome-free fragments are enriched at the TSSs, whereas the nucleosome-bound fragments should be enriched both upstream and downstream of the active TSSs and display characteristic phasing of upstream and downstream nucleosomes. Because ATAC-seq reads are concentrated at regions of open chromatin, users should see a strong nucleosome signal at the +1 nucleosome, but the signal decreases at the +2, +3 and +4 nucleosomes.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ChIPpeakAnno</span><span class="op">)</span>
<span class="va">bamfiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">outPath</span>,
                     <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NucleosomeFree.bam"</span>,
                     <span class="st">"mononucleosome.bam"</span>,
                     <span class="st">"dinucleosome.bam"</span>,
                     <span class="st">"trinucleosome.bam"</span><span class="op">)</span><span class="op">)</span>
<span class="co">## Plot the cumulative percentage of tag allocation in nucleosome-free </span>
<span class="co">## and mononucleosome bam files.</span>
<span class="fu"><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/cumulativePercentage.html">cumulativePercentage</a></span><span class="op">(</span><span class="va">bamfiles</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span>, <span class="fu">as</span><span class="op">(</span><span class="fu">seqinfo</span><span class="op">(</span><span class="va">Hsapiens</span><span class="op">)</span><span class="op">[</span><span class="st">"chr1"</span><span class="op">]</span>, <span class="st">"GRanges"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="ATACseqQC_files/figure-html/unnamed-chunk-13-1.png" width="384"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">TSS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html">promoters</a></span><span class="op">(</span><span class="va">txs</span>, upstream<span class="op">=</span><span class="fl">0</span>, downstream<span class="op">=</span><span class="fl">1</span><span class="op">)</span>
<span class="va">TSS</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">TSS</span><span class="op">)</span>
<span class="co">## estimate the library size for normalization</span>
<span class="op">(</span><span class="va">librarySize</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/estLibSize.html">estLibSize</a></span><span class="op">(</span><span class="va">bamfiles</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## splited/NucleosomeFree.bam splited/mononucleosome.bam 
##                      33384                       2139 
##   splited/dinucleosome.bam  splited/trinucleosome.bam 
##                       2030                        451</code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## calculate the signals around TSSs.</span>
<span class="va">NTILE</span> <span class="op">&lt;-</span> <span class="fl">101</span>
<span class="va">dws</span> <span class="op">&lt;-</span> <span class="va">ups</span> <span class="op">&lt;-</span> <span class="fl">1010</span>
<span class="va">sigs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enrichedFragments.html">enrichedFragments</a></span><span class="op">(</span>gal<span class="op">=</span><span class="va">objs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"NucleosomeFree"</span>, 
                                     <span class="st">"mononucleosome"</span>,
                                     <span class="st">"dinucleosome"</span>,
                                     <span class="st">"trinucleosome"</span><span class="op">)</span><span class="op">]</span>, 
                          TSS<span class="op">=</span><span class="va">TSS</span>,
                          librarySize<span class="op">=</span><span class="va">librarySize</span>,
                          seqlev<span class="op">=</span><span class="va">seqlev</span>,
                          TSS.filter<span class="op">=</span><span class="fl">0.5</span>,
                          n.tile <span class="op">=</span> <span class="va">NTILE</span>,
                          upstream <span class="op">=</span> <span class="va">ups</span>,
                          downstream <span class="op">=</span> <span class="va">dws</span><span class="op">)</span>
<span class="co">## log2 transformed signals</span>
<span class="va">sigs.log2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">sigs</span>, <span class="kw">function</span><span class="op">(</span><span class="va">.ele</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log2</a></span><span class="op">(</span><span class="va">.ele</span><span class="op">+</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span>
<span class="co">#plot heatmap</span>
<span class="fu"><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/featureAlignedHeatmap.html">featureAlignedHeatmap</a></span><span class="op">(</span><span class="va">sigs.log2</span>, <span class="fu"><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/reCenterPeaks.html">reCenterPeaks</a></span><span class="op">(</span><span class="va">TSS</span>, width<span class="op">=</span><span class="va">ups</span><span class="op">+</span><span class="va">dws</span><span class="op">)</span>,
                      zeroAt<span class="op">=</span><span class="fl">.5</span>, n.tile<span class="op">=</span><span class="va">NTILE</span><span class="op">)</span></code></pre></div>
<p><img src="ATACseqQC_files/figure-html/unnamed-chunk-14-1.png" width="384"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## get signals normalized for nucleosome-free and nucleosome-bound regions.</span>
<span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/featureAlignedDistribution.html">featureAlignedDistribution</a></span><span class="op">(</span><span class="va">sigs</span>, 
                                  <span class="fu"><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/reCenterPeaks.html">reCenterPeaks</a></span><span class="op">(</span><span class="va">TSS</span>, width<span class="op">=</span><span class="va">ups</span><span class="op">+</span><span class="va">dws</span><span class="op">)</span>,
                                  zeroAt<span class="op">=</span><span class="fl">.5</span>, n.tile<span class="op">=</span><span class="va">NTILE</span>, type<span class="op">=</span><span class="st">"l"</span>, 
                                  ylab<span class="op">=</span><span class="st">"Averaged coverage"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## rescale the nucleosome-free and nucleosome signals to 0~1</span>
<span class="va">range01</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="op">(</span><span class="va">x</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">}</span>
<span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span><span class="op">(</span><span class="va">out</span>, <span class="fl">2</span>, <span class="va">range01</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/matplot.html">matplot</a></span><span class="op">(</span><span class="va">out</span>, type<span class="op">=</span><span class="st">"l"</span>, xaxt<span class="op">=</span><span class="st">"n"</span>, 
        xlab<span class="op">=</span><span class="st">"Position (bp)"</span>, 
        ylab<span class="op">=</span><span class="st">"Fraction of signal"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/axis.html">axis</a></span><span class="op">(</span><span class="fl">1</span>, at<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">100</span>, by<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">+</span><span class="fl">1</span>, 
     labels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"-1K"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="op">-</span><span class="fl">800</span>, <span class="fl">800</span>, by<span class="op">=</span><span class="fl">200</span><span class="op">)</span>, <span class="st">"1K"</span><span class="op">)</span>, las<span class="op">=</span><span class="fl">2</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span>v<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">100</span>, by<span class="op">=</span><span class="fl">10</span><span class="op">)</span><span class="op">+</span><span class="fl">1</span>, lty<span class="op">=</span><span class="fl">2</span>, col<span class="op">=</span><span class="st">"gray"</span><span class="op">)</span></code></pre></div>
<p><img src="ATACseqQC_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
</div>
<div id="plot-footprints" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-footprints" class="anchor"></a>plot Footprints</h2>
<p>ATAC-seq footprints infer factor occupancy genome-wide. The <code>factorFootprints</code> function uses <code>matchPWM</code> to predict the binding sites using the input position weight matrix (PWM). Then it calculates and plots the accumulated coverage for those binding sites to show the status of the occupancy genome-wide. Unlike CENTIPEDE<span class="citation"><sup>4</sup></span>, the footprints generated here do not take the conservation (PhyloP) into consideration. <code>factorFootprints</code> function could also accept the binding sites as a GRanges object.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## foot prints</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">MotifDb</span><span class="op">)</span>
<span class="va">CTCF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MotifDb/man/query.html">query</a></span><span class="op">(</span><span class="va">MotifDb</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"CTCF"</span><span class="op">)</span><span class="op">)</span>
<span class="va">CTCF</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="va">CTCF</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">CTCF</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, digits<span class="op">=</span><span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>##      1    2    3     4     5     6     7     8     9    10    11    12    13
## A 0.10 0.16 0.30 0.072 0.012 0.786 0.024 0.122 0.914 0.012 0.376 0.022 0.028
## C 0.36 0.21 0.10 0.826 0.966 0.024 0.620 0.494 0.010 0.008 0.010 0.022 0.002
## G 0.12 0.41 0.44 0.050 0.012 0.108 0.336 0.056 0.048 0.976 0.602 0.606 0.962
## T 0.42 0.22 0.16 0.052 0.010 0.082 0.020 0.328 0.028 0.004 0.012 0.350 0.008
##      14    15    16    17   18   19
## A 0.024 0.096 0.424 0.086 0.12 0.34
## C 0.016 0.818 0.024 0.532 0.35 0.26
## G 0.880 0.038 0.522 0.326 0.12 0.32
## T 0.080 0.048 0.030 0.056 0.41 0.08</code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sigs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/factorFootprints.html">factorFootprints</a></span><span class="op">(</span><span class="va">shiftedBamfile</span>, pfm<span class="op">=</span><span class="va">CTCF</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, 
                         genome<span class="op">=</span><span class="va">genome</span>,
                         min.score<span class="op">=</span><span class="st">"90%"</span>, seqlev<span class="op">=</span><span class="va">seqlev</span>,
                         upstream<span class="op">=</span><span class="fl">100</span>, downstream<span class="op">=</span><span class="fl">100</span><span class="op">)</span></code></pre></div>
<p><img src="ATACseqQC_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/featureAlignedHeatmap.html">featureAlignedHeatmap</a></span><span class="op">(</span><span class="va">sigs</span><span class="op">$</span><span class="va">signal</span>, 
                      feature.gr<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/reCenterPeaks.html">reCenterPeaks</a></span><span class="op">(</span><span class="va">sigs</span><span class="op">$</span><span class="va">bindingSites</span>,
                                               width<span class="op">=</span><span class="fl">200</span><span class="op">+</span><span class="fu">width</span><span class="op">(</span><span class="va">sigs</span><span class="op">$</span><span class="va">bindingSites</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>, 
                      annoMcols<span class="op">=</span><span class="st">"score"</span>,
                      sortBy<span class="op">=</span><span class="st">"score"</span>,
                      n.tile<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">sigs</span><span class="op">$</span><span class="va">signal</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="ATACseqQC_files/figure-html/unnamed-chunk-18-1.png" width="576"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sigs</span><span class="op">$</span><span class="va">spearman.correlation</span></code></pre></div>
<pre><code>## $`+`
## 
##  Spearman's rank correlation rho
## 
## data:  predictedBindingSiteScore and highest.sig.windows
## S = 2483380, p-value = 1.454e-05
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##       rho 
## 0.2595536 
## 
## 
## $`-`
## 
##  Spearman's rank correlation rho
## 
## data:  predictedBindingSiteScore and highest.sig.windows
## S = 2697867, p-value = 0.001185
## alternative hypothesis: true rho is not equal to 0
## sample estimates:
##       rho 
## 0.1956021</code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sigs</span><span class="op">$</span><span class="va">Profile.segmentation</span></code></pre></div>
<pre><code>##           pos   distal_abun proximal_abun       binding 
##   55.75000000    0.09731340    0.14611895    0.05690943</code></pre>
<p>Here is the CTCF footprints for the full dataset. <img src="CTCFfootprints.png" alt="CTCF footprints"></p>
<div id="v-plot" class="section level3">
<h3 class="hasAnchor">
<a href="#v-plot" class="anchor"></a>V-plot</h3>
<p>V-plot is a plot to visualize fragment midpoint vs length for a given transcription factors.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">vp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/vPlot.html">vPlot</a></span><span class="op">(</span><span class="va">shiftedBamfile</span>, pfm<span class="op">=</span><span class="va">CTCF</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, 
            genome<span class="op">=</span><span class="va">genome</span>, min.score<span class="op">=</span><span class="st">"90%"</span>, seqlev<span class="op">=</span><span class="va">seqlev</span>,
            upstream<span class="op">=</span><span class="fl">200</span>, downstream<span class="op">=</span><span class="fl">200</span>, 
            ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">30</span>, <span class="fl">250</span><span class="op">)</span>, bandwidth<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="ATACseqQC_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/distanceDyad.html">distanceDyad</a></span><span class="op">(</span><span class="va">vp</span>, pch<span class="op">=</span><span class="fl">20</span>, cex<span class="op">=</span><span class="fl">.5</span><span class="op">)</span></code></pre></div>
<p><img src="ATACseqQC_files/figure-html/unnamed-chunk-19-2.png" width="700"></p>
<p>Here is the CTCF vPlot for the full dataset. <img src="CTCFvPlot.png" alt="CTCF footprints"></p>
</div>
</div>
</div>
<div id="plot-correlations-for-multiple-samples" class="section level1">
<h1 class="hasAnchor">
<a href="#plot-correlations-for-multiple-samples" class="anchor"></a>Plot correlations for multiple samples</h1>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package<span class="op">=</span><span class="st">"ATACseqQC"</span>, mustWork<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">bamfiles</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"*.bam$"</span>, full.name<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">gals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">bamfiles</span>, <span class="kw">function</span><span class="op">(</span><span class="va">bamfile</span><span class="op">)</span><span class="op">{</span>
               <span class="fu"><a href="../reference/readBamFile.html">readBamFile</a></span><span class="op">(</span>bamFile<span class="op">=</span><span class="va">bamfile</span>, tag<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">character</a></span><span class="op">(</span><span class="fl">0</span><span class="op">)</span>, 
                          which<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="fu">IRanges</span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1e6</span><span class="op">)</span><span class="op">)</span>, 
                          asMates<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>
         <span class="op">}</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span>
<span class="va">txs</span> <span class="op">&lt;-</span> <span class="fu">transcripts</span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicAlignments">GenomicAlignments</a></span><span class="op">)</span>
<span class="fu"><a href="../reference/plotCorrelation.html">plotCorrelation</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicAlignments/man/GAlignmentsList-class.html">GAlignmentsList</a></span><span class="op">(</span><span class="va">gals</span><span class="op">)</span>, <span class="va">txs</span>, seqlev<span class="op">=</span><span class="st">"chr1"</span><span class="op">)</span></code></pre></div>
<p><img src="ATACseqQC_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R Under development (unstable) (2021-03-18 r80099)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] GenomicAlignments_1.27.2               
##  [2] Rsamtools_2.7.1                        
##  [3] SummarizedExperiment_1.21.1            
##  [4] MatrixGenerics_1.3.1                   
##  [5] matrixStats_0.58.0                     
##  [6] MotifDb_1.33.0                         
##  [7] phastCons100way.UCSC.hg19_3.7.2        
##  [8] GenomicScores_2.3.4                    
##  [9] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2
## [10] GenomicFeatures_1.43.7                 
## [11] AnnotationDbi_1.53.1                   
## [12] Biobase_2.51.0                         
## [13] BSgenome.Hsapiens.UCSC.hg19_1.4.3      
## [14] BSgenome_1.59.2                        
## [15] rtracklayer_1.51.5                     
## [16] Biostrings_2.59.2                      
## [17] XVector_0.31.1                         
## [18] ChIPpeakAnno_3.25.4                    
## [19] GenomicRanges_1.43.3                   
## [20] GenomeInfoDb_1.27.8                    
## [21] IRanges_2.25.6                         
## [22] ATACseqQC_1.15.7                       
## [23] S4Vectors_0.29.9                       
## [24] BiocGenerics_0.37.1                    
## 
## loaded via a namespace (and not attached):
##   [1] AnnotationHub_2.23.2          BiocFileCache_1.15.1         
##   [3] systemfonts_1.0.1             lazyeval_0.2.2               
##   [5] splines_4.1.0                 BiocParallel_1.25.5          
##   [7] ggplot2_3.3.3                 digest_0.6.27                
##   [9] ensembldb_2.15.2              htmltools_0.5.1.1            
##  [11] fansi_0.4.2                   magrittr_2.0.1               
##  [13] memoise_2.0.0                 grImport2_0.2-0              
##  [15] InteractionSet_1.19.0         limma_3.47.12                
##  [17] askpass_1.1                   pkgdown_1.6.1                
##  [19] prettyunits_1.1.1             jpeg_0.1-8.1                 
##  [21] colorspace_2.0-0              blob_1.2.1                   
##  [23] rappdirs_0.3.3                textshaping_0.3.3            
##  [25] xfun_0.22                     dplyr_1.0.5                  
##  [27] crayon_1.4.1                  RCurl_1.98-1.3               
##  [29] jsonlite_1.7.2                graph_1.69.0                 
##  [31] survival_3.2-10               glue_1.4.2                   
##  [33] gtable_0.3.0                  zlibbioc_1.37.0              
##  [35] DelayedArray_0.17.10          Rhdf5lib_1.13.4              
##  [37] HDF5Array_1.19.10             scales_1.1.1                 
##  [39] futile.options_1.0.1          DBI_1.1.1                    
##  [41] edgeR_3.33.3                  Rcpp_1.0.6                   
##  [43] xtable_1.8-4                  progress_1.2.2               
##  [45] bit_4.0.4                     htmlwidgets_1.5.3            
##  [47] httr_1.4.2                    ellipsis_0.3.1               
##  [49] pkgconfig_2.0.3               XML_3.99-0.6                 
##  [51] sass_0.3.1                    dbplyr_2.1.0                 
##  [53] locfit_1.5-9.4                utf8_1.2.1                   
##  [55] polynom_1.4-0                 later_1.1.0.1                
##  [57] tidyselect_1.1.0              rlang_0.4.10                 
##  [59] munsell_0.5.0                 BiocVersion_3.13.1           
##  [61] tools_4.1.0                   cachem_1.0.4                 
##  [63] generics_0.1.0                RSQLite_2.2.5                
##  [65] ade4_1.7-16                   evaluate_0.14                
##  [67] stringr_1.4.0                 fastmap_1.1.0                
##  [69] yaml_2.2.1                    ragg_1.1.2                   
##  [71] knitr_1.31                    bit64_4.0.5                  
##  [73] fs_1.5.0                      randomForest_4.6-14          
##  [75] purrr_0.3.4                   splitstackshape_1.4.8        
##  [77] KEGGREST_1.31.1               AnnotationFilter_1.15.0      
##  [79] RBGL_1.67.0                   mime_0.10                    
##  [81] formatR_1.8                   xml2_1.3.2                   
##  [83] biomaRt_2.47.6                BiocStyle_2.19.2             
##  [85] debugme_1.1.0                 compiler_4.1.0               
##  [87] filelock_1.0.2                curl_4.3                     
##  [89] png_0.1-7                     interactiveDisplayBase_1.29.0
##  [91] preseqR_4.0.0                 tibble_3.1.0                 
##  [93] bslib_0.2.4                   stringi_1.5.3                
##  [95] highr_0.8                     futile.logger_1.4.3          
##  [97] desc_1.3.0                    lattice_0.20-41              
##  [99] ProtGenerics_1.23.7           Matrix_1.3-2                 
## [101] multtest_2.47.0               vctrs_0.3.7                  
## [103] pillar_1.5.1                  lifecycle_1.0.0              
## [105] rhdf5filters_1.3.4            BiocManager_1.30.12          
## [107] jquerylib_0.1.3               data.table_1.14.0            
## [109] bitops_1.0-6                  httpuv_1.5.5                 
## [111] R6_2.5.0                      BiocIO_1.1.2                 
## [113] promises_1.2.0.1              KernSmooth_2.23-18           
## [115] motifStack_1.35.0             lambda.r_1.2.4               
## [117] MASS_7.3-53.1                 assertthat_0.2.1             
## [119] rhdf5_2.35.2                  openssl_1.4.3                
## [121] rprojroot_2.0.2               rjson_0.2.20                 
## [123] regioneR_1.23.0               GenomeInfoDbData_1.2.4       
## [125] hms_1.0.0                     VennDiagram_1.6.20           
## [127] grid_4.1.0                    rmarkdown_2.7                
## [129] base64enc_0.1-3               shiny_1.6.0                  
## [131] restfulr_0.0.13</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body" line-spacing="2">
<div id="ref-buenrostro2013transposition" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Buenrostro, J. D., Giresi, P. G., Zaba, L. C., Chang, H. Y. &amp; Greenleaf, W. J. Transposition of native chromatin for fast and sensitive epigenomic profiling of open chromatin, DNA-binding proteins and nucleosome position. <em>Nature methods</em> <strong>10,</strong> 1213–1218 (2013).</div>
</div>
<div id="ref-adey2010rapid" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Adey, A. <em>et al.</em> Rapid, low-input, low-bias construction of shotgun fragment libraries by high-density in vitro transposition. <em>Genome biology</em> <strong>11,</strong> R119 (2010).</div>
</div>
<div id="ref-chen2013danpos" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">Chen, K. <em>et al.</em> DANPOS: Dynamic analysis of nucleosome position and occupancy by sequencing. <em>Genome research</em> <strong>23,</strong> 341–351 (2013).</div>
</div>
<div id="ref-pique2011accurate" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">Pique-Regi, R. <em>et al.</em> Accurate inference of transcription factor binding from DNA sequence and chromatin accessibility data. <em>Genome research</em> <strong>21,</strong> 447–455 (2011).</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Jianhong Ou.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
